### DISCUSSION TITLE
Giving error squiggles on things that should not have error squiggles

### DISCUSSION THREAD

**Discussion by Zephxr** [microsoft/vscode-cpptools#9896] [Q&A]
On my laptop, I use the application flawlessly and I get perfect squiggles on bad code, but on my desktop, I have varied results.
For example, this code runs fine on both machines:
![image](https://user-images.githubusercontent.com/71910018/191133754-deca3972-ddd9-4124-897c-b1e24a5c365c.png)
![image](https://user-images.githubusercontent.com/71910018/191133759-e6266a55-2b8e-4dd7-912d-ec1cd3decd8c.png)
however, only on my desktop will I get the squiggles on this code.  Anyone know how to fix this?  I determined it was this extension because it does not show up if I disable it.

**Comment by sean-mcmanus** [comment]
What is the reported error? We need info on how IntelliSense is being configured, such as what is reported from running C/C++: Log Diagnostics. It may be running a different IntelliSense mode.

**Reply by Zephxr** [reply]
![image](https://user-images.githubusercontent.com/71910018/192868786-d2053e26-f67a-4c7e-bf62-2a42ac66331e.png)
After my volatile, it will always give an error of "expected ;"
![image](https://user-images.githubusercontent.com/71910018/192869013-69eb8721-0249-485b-a847-1deda267e81a.png)
After my asm, it will give "expected a type specifier"

**Reply by sean-mcmanus** [reply]
We need to know how your IntelliSense is being configured. e.g. from C/C++: Log Diagnostics, i.e. compilerPath, IntelliSenseMode, etc.

**Reply by Zephxr** [reply]
Sorry, I'm new to this.  How should I go about sharing this information and how should I get to it?

**Reply by sean-mcmanus** [reply]
The C/C++: Log Diagnostics command is available from the View->Command Palette. Can you add the info (or at least of subset of it) in this thread?

**Reply by Zephxr** [reply]
This is what I got
`-------- Diagnostics - 10/17/2022, 3:23:22 PM
Version: 1.12.4
Current Configuration:
{
    "name": "Win32",
    "includePath": [
        "${workspaceFolder}/**"
    ],
    "defines": [
        "_DEBUG",
        "UNICODE",
        "_UNICODE"
    ],
    "windowsSdkVersion": "10.0.19041.0",
    "compilerPath": "C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.33.31629/bin/Hostx64/x64/cl.exe",
    "cStandard": "c17",
    "cppStandard": "c++17",
    "intelliSenseMode": "windows-msvc-x64",
    "intelliSenseModeIsExplicit": false,
    "cStandardIsExplicit": false,
    "cppStandardIsExplicit": false,
    "mergeConfigurations": false,
    "compilerPathIsExplicit": false,
    "browse": {
        "path": [
            "${workspaceFolder}/**"
        ],
        "limitSymbolsToIncludedHeaders": true
    }
}
Translation Unit Mappings:
[ C:\Users\gavin\Documents\Github Repo's\OS\etec3701\interrupt.c ]:
    C:\Users\gavin\Documents\Github Repo's\OS\etec3701\interrupt.c
Translation Unit Configurations:
[ C:\Users\gavin\Documents\Github Repo's\OS\etec3701\interrupt.c ]:
    Process ID: 16692
    Memory Usage: 16 MB
    Compiler Path: C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.33.31629/bin/Hostx64/x64/cl.exe
    Includes:
        C:\PROGRAM FILES\MICROSOFT VISUAL STUDIO\2022\COMMUNITY\VC\TOOLS\MSVC\14.33.31629\INCLUDE
        C:\PROGRAM FILES\MICROSOFT VISUAL STUDIO\2022\COMMUNITY\VC\TOOLS\MSVC\14.33.31629\ATLMFC\INCLUDE
        C:\PROGRAM FILES (X86)\WINDOWS KITS\10\INCLUDE\10.0.19041.0\UM
        C:\PROGRAM FILES (X86)\WINDOWS KITS\10\INCLUDE\10.0.19041.0\UCRT
        C:\PROGRAM FILES (X86)\WINDOWS KITS\10\INCLUDE\10.0.19041.0\SHARED
        C:\PROGRAM FILES (X86)\WINDOWS KITS\10\INCLUDE\10.0.19041.0\WINRT
        C:\PROGRAM FILES (X86)\WINDOWS KITS\10\INCLUDE\10.0.19041.0\CPPWINRT
    Defines:
        _DEBUG
        UNICODE
        _UNICODE
    Standard Version: ms_c17
    IntelliSense Mode: windows-msvc-x64
Total Memory Usage: 16 MB

------- Workspace parsing diagnostics -------
Number of files discovered (not excluded): 4911`

**Reply by sean-mcmanus** [reply]
@Zephxr Your compilerPath is set to cl.exe and "asm volatile" doesn't compile with that compiler. Can you change your compilerPath to the one you're compiling with?

**Reply by Zephxr** [reply]
Sorry for the late reply, how would I do that?

**Reply by sean-mcmanus** [reply]
You could open the c_cpp_properties.json file and change the compilerPath property to match the path to your compiler.
